core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -610,6 +609,17 @@ class TransactionStateManager(brokerId: Int,
     }
   }
 
+  /**
+   * Remove the loading partition if the epoch is less than the specified epoch. Note: This method must be called under the write state lock.
+   */
+  private def removeLoadingPartitionWithEpoch(partitionId: Int, coordinatorEpoch: Int): Unit = {
+    loadingPartitions.find(_.txnPartitionId == partitionId).foreach { partitionAndLeaderEpoch =>
+      if (partitionAndLeaderEpoch.coordinatorEpoch < coordinatorEpoch) {
+        loadingPartitions.remove(partitionAndLeaderEpoch)
```

Comment:
I think this should be pretty rare, so we could add an info level logging.

Reply:
We just want to say we removed a loading partition?

Reply:
Something along the lines of "Cancelling load of currently loading partition $partitionAndLeaderEpoch".